src/components/dock/dock.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    h,
+    Prop,
+} from '@stencil/core';
+import { DockItem } from './dock.types';
+
+const DEFAULT_MOBILE_BREAKPOINT = 700;
+
+/**
+ * @exampleComponent limel-example-dock-basic
+ * @exampleComponent limel-example-dock-mobile
+ * @private
+ */
+@Component({
+    tag: 'limel-dock',
+    shadow: true,
+    styleUrl: 'dock.scss',
+})
+export class Dock {
+    @Element()
+    public element: HTMLLimelDockElement;
+
+    /**
+     * What dock items to render
+     */
+    @Prop()
+    public dockItems: DockItem[] = [];
+
+    /**
+     * Fired when a dock item has been selected from the dock
+     */
+    @Event()
+    public change: EventEmitter<DockItem>;
+
+    /**
+     * Defines the width of the component and whether it should
+     * be wide and show the labels of the dock items,
+     * or be small and only show icons of the items.
+     */
+    @Prop({ reflect: true })
+    public expanded = false;
+
+    /**
+     * xxxxxx
+     */
+    @Prop({ reflect: true })
+    public mobileBreakPoint: number = DEFAULT_MOBILE_BREAKPOINT;
+
+    /**
+     * xxxxxx
+     */
+    private hasMobileLayout = false;
+
+    public componentWillLoad() {
+        if (window.innerWidth <= this.mobileBreakPoint) {
```

Comment:
**⚠️ for the reviewer:** the reason we couldn't add CSS variables for consumers to define their mobile breakpoint is this: https://bholmes.dev/blog/alternative-to-css-variable-media-queries/